src/data/finset/basic.lean

```diff
@@ -1447,12 +1453,60 @@ lemma union_sdiff_self (s t : finset α) : (s ∪ t) \ t = s \ t := sup_sdiff_ri
 lemma sdiff_singleton_eq_erase (a : α) (s : finset α) : s \ singleton a = erase s a :=
 by { ext, rw [mem_erase, mem_sdiff, mem_singleton], tauto }
 
-@[simp] lemma sdiff_singleton_not_mem_eq_self (s : finset α) {a : α} (ha : a ∉ s) : s \ {a} = s :=
-by simp only [sdiff_singleton_eq_erase, ha, erase_eq_of_not_mem, not_false_iff]
+lemma erase_eq (s : finset α) (a : α) : s.erase a = s \ {a} := (sdiff_singleton_eq_erase _ _).symm
+
+lemma disjoint_erase_comm : disjoint (s.erase a) t ↔ disjoint s (t.erase a) :=
+by simp_rw [erase_eq, disjoint_sdiff_comm]
+
+lemma disjoint_of_erase_left (ha : a ∉ t) (hst : disjoint (s.erase a) t) : disjoint s t :=
+by { rw [←erase_insert ha, ←disjoint_erase_comm, disjoint_insert_right],
+  exact ⟨not_mem_erase _ _, hst⟩ }
+
+lemma disjoint_of_erase_right (ha : a ∉ s) (hst : disjoint s (t.erase a)) : disjoint s t :=
+by { rw [←erase_insert ha, disjoint_erase_comm, disjoint_insert_left],
+  exact ⟨not_mem_erase _ _, hst⟩ }
+
+@[simp] lemma inter_erase (a : α) (s t : finset α) : s ∩ t.erase a = (s ∩ t).erase a :=
+by simp only [erase_eq, inter_sdiff]
+
+@[simp] lemma erase_inter (a : α) (s t : finset α) : s.erase a ∩ t = (s ∩ t).erase a :=
+by simpa only [inter_comm t] using inter_erase a t s
+
+lemma erase_sdiff_comm (s t : finset α) (a : α) : s.erase a \ t = (s \ t).erase a :=
+by simp_rw [erase_eq, sdiff_right_comm]
+
+lemma insert_union_comm (s t : finset α) (a : α) : insert a s ∪ t = s ∪ insert a t :=
+by rw [insert_union, union_insert]
```

Comment:
`comm` seems to mean `bin (unary x) y = unary (bin x y)` in the first name but `bin (unary x) y = bin x (unary y)` in the second one, Calling the first one `erase_sdiff` would ease this a little.

Reply:
If you consider both `insert` and `erase` to be binary, they're respectively
* `bin1 (bin2 s a) t = bin2 (bin1 s t) a`
* `bin1 (bin2 a s) t = bin1 s (bin2 a t)`

so the first one is clearly right commutativity, but the second one isn't clearly anything.